_inc/lib/widgets.php

```diff
@@ -120,6 +120,23 @@ public static function get_widget_by_id( $widget_id ) {
 		return $found;
 	}
 
+	/**
+	 * Return a widget by base ID or null if nothing is found.
+	 *
+	 * @param string $id_base The id of a widget to look for.
+	 *
+	 * @return array|null The matching formatted widget (see format_widget).
+	 */
+	public static function get_widget_by_id_base( $id_base ) {
+		$found = null;
+		foreach ( self::get_all_widgets() as $widget ) {
+			if ( $widget['id_base'] === $id_base ) {
+				$found = $widget;
```

Comment:
If we alter the `return $found` to `return null` and we alter `$found = $widget` to `return $widget`, we can omit `$found = null` at the top.